sequencer/src/state_signature/relay_server.rs

```diff
@@ -0,0 +1,259 @@
+use crate::state_signature::StateSignatureScheme;
+
+use super::{LightClientState, StateSignatureRequestBody};
+use async_compatibility_layer::channel::OneShotReceiver;
+use async_std::sync::RwLock;
+use clap::Args;
+use ethers::types::U256;
+use futures::FutureExt;
+use hotshot_stake_table::vec_based::config::FieldType;
+use hotshot_types::light_client::{StateSignature, StateVerKey};
+use jf_primitives::signatures::SignatureScheme;
+use jf_utils::to_bytes;
+use serde::{Deserialize, Serialize};
+use std::{
+    collections::{BTreeSet, HashMap},
+    path::PathBuf,
+};
+use tide_disco::{
+    api::ApiError,
+    error::ServerError,
+    method::{ReadState, WriteState},
+    Api, App, Error as _, StatusCode,
+};
+use url::Url;
+
+/// The state signatures package is a light client state and its signatures collected
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub struct StateSignaturesPackage {
+    /// The state for this signatures package
+    state: LightClientState,
+    /// The collected signatures
+    signatures: HashMap<StateVerKey, StateSignature>,
+
+    /// Total stakes associated with the signer
+    accumulated_weight: U256,
+}
+
+/// State that checks the light client state update and the signature collection
+#[derive(Default)]
+struct StateRelayServerState {
+    /// Minimum weight to form an available state signature package
+    threshold: U256,
+    /// Stake table
+    known_nodes: HashMap<StateVerKey, U256>,
+    /// Signatures packages for each block height
+    packages: HashMap<u64, StateSignaturesPackage>,
+
+    /// The latest state signatures package whose total weight exceeds the threshold
+    latest_available_package: Option<StateSignaturesPackage>,
```

Comment:
Could we remove this? Is it equivalent to `packages[latest_block_height]`?

Reply:
I'm hoping in the future we could push `RwLock` inside this state struct. So that when nodes are submitting signatures and locking `packages`, prover could still read the latest package here.

Reply:
Ah,  good idea